From 5a5cffd99aa441b127d004e5745d28d675f6f5f4 Mon Sep 17 00:00:00 2001
From: Alex Marginean <alexandru.marginean@nxp.com>
Date: Thu, 22 Aug 2019 12:47:12 +0300
Subject: [PATCH 642/741] arm64: dts: LS1028a-rdb: use Ethernet PHY interrupt

commit 07fa03bbd96a7c945e0e9345d1cdba56be628246 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

Use the PHY interrupt wired to GPIO pins as part of MDIO WA performance
impact mitigation.

Signed-off-by: Alex Marginean <alexandru.marginean@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts b/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts
index c72eb5c4b538..08d482ae5531 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts
@@ -200,6 +200,8 @@
 		#size-cells = <0>;
 		sgmii_phy0: ethernet-phy@2 {
 			reg = <0x2>;
+			interrupt-parent = <&gpio1>;
+			interrupts = <25 IRQ_TYPE_EDGE_FALLING>;
 		};
 	};
 };
@@ -211,18 +213,26 @@
 &enetc_mdio_pf3 {
 	qsgmii_phy1: ethernet-phy@4 {
 		reg = <0x10>;
+		interrupt-parent = <&gpio1>;
+		interrupts = <25 IRQ_TYPE_EDGE_FALLING>;
 	};
 
 	qsgmii_phy2: ethernet-phy@5 {
 		reg = <0x11>;
+		interrupt-parent = <&gpio1>;
+		interrupts = <25 IRQ_TYPE_EDGE_FALLING>;
 	};
 
 	qsgmii_phy3: ethernet-phy@6 {
 		reg = <0x12>;
+		interrupt-parent = <&gpio1>;
+		interrupts = <25 IRQ_TYPE_EDGE_FALLING>;
 	};
 
 	qsgmii_phy4: ethernet-phy@7 {
 		reg = <0x13>;
+		interrupt-parent = <&gpio1>;
+		interrupts = <25 IRQ_TYPE_EDGE_FALLING>;
 	};
 };
 
-- 
2.17.1

